k8s/controllers/solution/instance_controller.go

```diff
@@ -235,9 +240,11 @@ func (r *InstanceReconciler) handleSolution(obj client.Object) []ctrl.Request {
 	ret := make([]ctrl.Request, 0)
 	solObj := obj.(*solution_v1.Solution)
 	var instances solution_v1.InstanceList
+
+	solutionName := utils.ReplaceLastSeperator(solObj.Name, "-", ":")
```

Comment:
This makes me think whey we don't support "-" in object names? Let's say if user defines a "my-solution-v1" SolutionContainer, then the v1 Solution would be "my-solution-v1-v1" and be referenced as "my-solution-v1:v1". Is the concern that user will define a "my-solution-v1" Solution directly, which we don't aim to support?

Reply:
To avoid one conflicting scenario, customer A creates container "a-b" and create version "c". And another customer B creates contains "a" and version "b-c". These two cases create object "a-b-c" in k8s.  

Currently, we set naming constraints in ARM spec. No "-" or ". " is allowed in container name or version name. So the request from ARM contains only one "-" in the object name. We will do the same validation in webhook for the naming check to block customer creating with name having multiple "-". Once we release the naming constraint in ARM like allowing "-" in container name, we can do the same.